### PR TITLE
Adding `DDGate` to `qiskit_superstaq.custom_gates`

### PR DESCRIPTION
**Pull request by pgoiporia**

### CODE REVIEW

File: qiskit-superstaq/qiskit_superstaq/custom_gates.py [hunk 1]
```diff
@@ -507,3 +507,78 @@ def __init__(self, label: str | None = None) -> None:
...
+        return DDGate(-self.params[0])
+
+    def _define(self) -> None:
+        """Stores the qiskit circuit definition of the DD gate."""
+        qc = qiskit.QuantumCircuit(2, name="dd")
+        qc.rzz(2 * self.params[0], 0, 1)
+        qc.append(qiskit.circuit.library.XXPlusYYGate(-2 * self.params[0], 0), [0, 1])
+        self.definition = qc
+
+    def __array__(self, dtype: type | None = None) -> npt.NDArray[np.bool_]:
```
Comment (richrines1) [inline qiskit-superstaq/qiskit_superstaq/custom_gates.py:554]:
```suggestion
    def __array__(self, dtype: type | None = None) -> npt.NDArray[np.complex_]:
```
(this should be fixed for all the existing gates too)

File: qiskit-superstaq/qiskit_superstaq/custom_gates.py [hunk 2]
```diff
@@ -507,3 +507,78 @@ def __init__(self, label: str | None = None) -> None:
...
+                    np.exp(1j * self.params[0]) * np.cos(self.params[0]),
+                    0,
+                ],
+                [0, 0, 0, np.exp(-1j * self.params[0])],
+            ],
+            dtype=dtype,
+        )
+
+    def __repr__(self) -> str:
+        args = f"{self.params[0]}"
```
Comment (richrines1) [inline qiskit-superstaq/qiskit_superstaq/custom_gates.py:577]:
```suggestion
        args = f"{self.params[0]!r}"
```

File: qiskit-superstaq/qiskit_superstaq/custom_gates_test.py [hunk 3]
```diff
@@ -219,3 +219,17 @@ def test_aqticcx() -> None:
     )
 
     np.allclose(qiskit.quantum_info.Operator(qc), correct_unitary)
+
+
+def test_dd_swap() -> None:
```
Comment (richrines1) [inline qiskit-superstaq/qiskit_superstaq/custom_gates_test.py:224]:
```suggestion
def test_dd_gate() -> None:
```

**Review by richrines1 — COMMENTED** [review]
looks great, just a few tweaks

File: qiskit-superstaq/qiskit_superstaq/custom_gates.py (hunk 1 above)
Comment (pgoiporia) [reply qiskit-superstaq/qiskit_superstaq/custom_gates.py:554]: Creating an issue for this. It's a very easy fix but out of the scope for this PR. Will assign to myself.

**Review by richrines1 — APPROVED** [review]
lgtm!

**Pull request merged**